onnxruntime/core/providers/coreml/builders/impl/builder_utils.cc

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <core/providers/common.h>
+
+#include "builder_utils.h"
+#include "coreml/NeuralNetwork.pb.h"
+#include "core/providers/coreml/builders/helper.h"
+
+namespace onnxruntime {
+namespace coreml {
+
+common::Status CreateCoreMLWeight(CoreML::Specification::WeightParams& weight,
+                                  const ONNX_NAMESPACE::TensorProto& tensor) {
+  auto data_type = tensor.data_type();
+  if (data_type = ONNX_NAMESPACE::TensorProto_DataType_FLOAT) {
+    const float* data =
+        tensor.float_data().empty() ? reinterpret_cast<const float*>(tensor.raw_data().data())
+                                    : tensor.float_data().data();
+
+    weight.mutable_floatvalue()->Clear();
+    auto num_elements = Product(tensor.dims());
+    std::copy(data, data + num_elements,
+              google::protobuf::RepeatedFieldBackInserter(weight.mutable_floatvalue()));
```

Comment:
Not sure what tags google::protobuf::RepeatedFieldBackInserter has and whether copy_n would allow this to happen in one block. may translate to using copy_n automatically if possible though.

Reply:
This part I am not exactly sure, I think the compiler will optimize these into copy_n or memcpy


---
In reply to: [563462675](https://github.com/microsoft/onnxruntime/pull/6392#discussion_r563462675) [](ancestors = 563462675)